### PR TITLE
Defer 'Store#emitChange' until promise resolve

### PR DESCRIPTION
**Pull request by mehcode**

### CODE REVIEW

**Comment by goatslacker** [comment]
This is fairly clean. Some initial thoughts: this needs to check if Promise exists, this only works for one store?

File: src/alt.js [hunk 1]
```diff
@@ -79,8 +79,12 @@ class AltStore {
       if (model[LISTENERS][payload.action]) {
         this[SET_STATE] = false
         const result = model[LISTENERS][payload.action](payload.data)
-        if (result !== false && this[SET_STATE] === false) {
-          this.emitChange()
+        if (this[SET_STATE] === false) {
+          if (result instanceof Promise) {
```
Comment (goatslacker) [inline src/alt.js]: I've included this change in https://github.com/goatslacker/alt/pull/234

Comment (mehcode) [inline src/alt.js]: Looks awesome; thanks for taking this and making it much better. :+1:

Comment (mull) [inline src/alt.js]: I know this is an old commit that doesn't seem that it'll get merged it. Just an FYI that `object instanceof Promise` fails on Chrome for native promises.

Comment (goatslacker) [inline src/alt.js]: :+1: we have an isPromise check.

**Pull request closed without merging**